bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -230,17 +232,19 @@ protected void removeModuleHandlerFactory(ModuleHandlerFactory moduleHandlerFact
      */
     @Override
     public Rule add(Rule rule) {
-        if (rule == null) {
-            throw new IllegalArgumentException("The added rule must not be null!");
-        }
         String rUID = rule.getUID();
         if (rUID == null) {
             rUID = ruleEngine.getUniqueId();
             super.add(initRuleId(rUID, rule));
         } else {
             super.add(rule);
         }
-        return get(rUID);
+        Rule ruleCopy = get(rUID);
```

Comment:
> as we saw that @nonnull does not defends us

It actually defends us especially on internal methods, because we have checks in place in our IDE and the Maven build. So we can be sure that all ESH code itself respects the annotations - imho definitely helpful for any kind of future refactorings.